### PR TITLE
Upgrade to Lit 2

### PR DESCRIPTION
**Pull request by gaetanmaisse**
**Issue**

NA

**Description**

New try after https://github.com/gravitee-io/gravitee-ui-components/pull/466

Upgrade to Lit 2:
- Smaller bundle size
- Better performance
- Top of the art JS code

### CODE REVIEW

**Comment by graviteeio** [comment]
Hey dear reviewer, I'm the Gravitee.io bot :robot:

I just want to tell you that `package.json` has been updated in this PR. 

Some components are based on third party libraries, for integration into a web component, we extract the css from its libraries as assets.

WARNING: When `codemirror`, `highlight.js` or `github-markdown-css` are updated, we must run `npm run build` task to update the css.

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 33f2175efa0f1fe541924c6be76cb302ac9a8c6b
</p>

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

**This is still experimental ⚠️**

To use it in your project run the following lines:
```shell
echo @gravitee-io:registry=https://npm.pkg.github.com > .npmrc
npm install @gravitee-io/ui-components@3.23.0-33f2175
rm -rf node_modules/@gravitee
mv node_modules/@gravitee-io node_modules/@gravitee
```

**Comment by gcusnieux** [comment]
@gaetanmaisse Did you reproduce the problems with the svg ?

**Comment by gaetanmaisse** [comment]
In fact there are some issues with schema form, I tried to investigate but didn't find anything, would be great if you can take a look or @ThibaudAV

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

**This is still experimental ⚠️**

To use it in your project run the following lines:
```shell
echo @gravitee-io:registry=https://npm.pkg.github.com > .npmrc
npm install @gravitee-io/ui-components@3.23.0-615ab6b
rm -rf node_modules/@gravitee
mv node_modules/@gravitee-io node_modules/@gravitee
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

**This is still experimental ⚠️**

To use it in your project run the following lines:
```shell
echo @gravitee-io:registry=https://npm.pkg.github.com > .npmrc
npm install @gravitee-io/ui-components@3.23.0-ad2bb8d
rm -rf node_modules/@gravitee
mv node_modules/@gravitee-io node_modules/@gravitee
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

**This is still experimental ⚠️**

To use it in your project run the following lines:
```shell
echo @gravitee-io:registry=https://npm.pkg.github.com > .npmrc
npm install @gravitee-io/ui-components@3.23.0-5f76646
rm -rf node_modules/@gravitee/ui-components
mv node_modules/@gravitee-io/ui-components node_modules/@gravitee/ui-components
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.23.0-lit-2-le-retour-8b7e4a0
```

**Review by ThibaudAV — APPROVED** [review]

**Comment by gaetanmaisse** [comment]
@gcusnieux it looks like the icon issue has been fixed, maybe in the latest version of lit 2 😄

File: src/atoms/gv-select-native.js [hunk 1]
```diff
@@ -182,7 +182,7 @@ export class GvSelectNative extends InputElement(LitElement) {
   }
 
   getInputElement() {
-    return this.shadowRoot.querySelector('select');
+    return super.getInputElement();
```
Comment (gcusnieux) [inline src/atoms/gv-select-native.js:185]: @ThibaudAV @gaetanmaisse It's chelou

File: src/organisms/gv-schema-form.js [hunk 2]
```diff
@@ -286,6 +287,7 @@ export class GvSchemaForm extends LitElement {
     const isWriteOnly = control.writeOnly === true;
     const value = get(this._values, key);
     return html`<gv-schema-form-control
+      data-a="dd"
```
Comment (gcusnieux) [inline src/organisms/gv-schema-form.js:290]: to remove

File: src/organisms/gv-schema-form.js [hunk 3]
```diff
@@ -393,7 +395,7 @@ export class GvSchemaForm extends LitElement {
     }
     const keys = this.schema.properties ? Object.keys(this.schema.properties) : [];
     this._ignoreProperties = [];
-    return html`${keys.map((key) => this._renderControl(key))}`;
+    return repeat(keys, (key) => this._renderControl(key));
```
Comment (gcusnieux) [inline src/organisms/gv-schema-form.js:398]: Be carefull with this, did you test a form validation in design studio ?

File: src/atoms/gv-select-native.js (hunk 1 above)
Comment (gaetanmaisse) [reply src/atoms/gv-select-native.js:185]: The implementation of this method was  the same as the one super class (our version is less protective). Maybe we should just remove the method to inherit the super one directly?

File: src/organisms/gv-schema-form.js (hunk 3 above)
Comment (gaetanmaisse) [reply src/organisms/gv-schema-form.js:398]: Thanks for the tip, I tried in schema forms but not in policy studio, will do it ;)

File: src/atoms/gv-select-native.js (hunk 1 above)
Comment (gcusnieux) [reply src/atoms/gv-select-native.js:185]: No it's not the same

**Pull request merged**